File: flutter-sdk/lib/src/data/repositories/image/image_repository.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:chat_flutter_sdk/src/common/result.dart';
import 'package:chat_flutter_sdk/src/domain/models/image/image_data.dart';

enum ImagePickSource { gallery, camera }

abstract class ImageRepository {
  Future<Result<ImageData?>> pickImage(ImagePickSource source);

  Future<Result<ImageData>> saveImage(ImageData imageData);

  Future<Result<Unit>> deleteImage(ImageData imageData);
}

File: android-sdk/sdk/src/main/java/ai/yalo/chat/sdk/data/repositories/image/ImageRepository.kt
// Copyright (c) Yalochat, Inc. All rights reserved.
package ai.yalo.chat.sdk.data.repositories.image

import ai.yalo.chat.sdk.domain.models.ImageAttachment
import android.net.Uri

/**
 * The pictures a person sends, on their way from the gallery into the
 * conversation.
 *
 * Nothing that picks a picture has to know where the copy the chat shows is
 * kept, or that the gallery may stop answering for the one that was picked.
 */
internal interface ImageRepository {

    /**
     * Takes a copy of what [uri] points at and describes it, ready to be stored
     * and sent. A picture the device will not hand over comes back as a failed
     * [Result], and nothing is left on disk.
     */
    suspend fun store(uri: Uri): Result<ImageAttachment>
}
